exporter/prometheusremotewriteexporter/helper.go

```diff
@@ -215,21 +220,47 @@ func getPromMetricName(metric *otlp.Metric, ns string) string {
 	return sanitize(b.String())
 }
 
-// Simple helper function that takes the <Signature String - *TimeSeries> map
-// and creates a WriteRequest from the struct -- can move to the helper.go file
-func wrapTimeSeries(tsMap map[string]*prompb.TimeSeries) (*prompb.WriteRequest, error) {
+// Helper function that takes the <Signature String - *TimeSeries> map
+// and creates a batch of WriteRequest from the struct
+func wrapAndBatchTimeSeries(tsMap map[string]*prompb.TimeSeries) ([]*prompb.WriteRequest, error) {
 	if len(tsMap) == 0 {
 		return nil, errors.New("invalid tsMap: cannot be empty map")
 	}
+
+	var requests []*prompb.WriteRequest
 	var tsArray []prompb.TimeSeries
+	missingLastBatch := true
+
 	for _, v := range tsMap {
 		tsArray = append(tsArray, *v)
+		missingLastBatch = true
+
+		if len(tsArray)%checkByteSizeInterval == 0 {
+			wrapped := prompb.WriteRequest{
+				Timeseries: tsArray,
+				// Other parameters of the WriteRequest are unnecessary for our Export
+			}
+
+			if b := proto.Size(&wrapped); b >= maxBatchByteSize || len(tsArray) >= maxBatchMetric {
```

Comment:
Why do we need to decided on if we need to create a new back base on `maxBatchByteSize` or `tsArray` length?

Why can't it be **either** `maxBatchByteSize` **or** `tsArray` length?

Reply:
Hm, we can definitely choose to use **either** maxBatchByteSize **or** tsArray. The reason why I used both was because I thought it would be the best way to maximize the byte size of each request efficiently. This is because we could avoid marshalling requests with a huge amount of metrics just to check the byte size. However, I realize that we actually limit the size of the objects we marshal already (through the batching).

I'll ask around to see which would be the better option.